modules/ducktests/tests/ignitetest/tests/client_in_out_test.py

```diff
@@ -0,0 +1,173 @@
+
+        servers_count = cluster - static_clients_num - temp_client
+
+        current_top_v = servers_count
+        # Topology version after test.
+        fin_top_ver = servers_count + (2 * static_clients_num) + (2 * iteration_count * temp_client)
+
+        server_cfg = IgniteConfiguration(
+            version=IgniteVersion(ignite_version),
+            caches=[CacheConfiguration(name=self.CACHE_NAME, backups=1, atomicity_mode='TRANSACTIONAL')]
+        )
+
+        ignite = IgniteService(self.test_context, server_cfg, num_nodes=servers_count)
+        control_utility = ControlUtility(ignite, self.test_context)
+
+        client_cfg = server_cfg._replace(client_mode=True)
+
+        static_clients = IgniteApplicationService(
+            self.test_context,
+            client_cfg,
+            java_class_name=self.JAVA_CLIENT_CLASS_NAME,
+            num_nodes=static_clients_num,
+            params={"cacheName": self.CACHE_NAME,
+                    "pacing": self.PACING})
+
+        temp_clients = IgniteApplicationService(
+            self.test_context,
+            client_cfg,
+            java_class_name=self.JAVA_CLIENT_CLASS_NAME,
+            num_nodes=temp_client,
+            params={"cacheName": self.CACHE_NAME,
+                    "pacing": self.PACING})
+
+        ignite.start()
+
+        static_clients.start()
+
+        current_top_v += static_clients_num
+        check_topology(control_utility, current_top_v)
+
+        # Start / stop temp_clients node. Check cluster.
+        for i in range(iteration_count):
+            self.logger.debug(f'Starting iteration: {i}.')
+
+            temp_clients.start()
+            current_top_v += temp_client
+
+            static_clients.await_event(f'ver={current_top_v}, locNode=', timeout_sec=80,
+                                       from_the_beginning=True, backoff_sec=1)
+            check_topology(control_utility, current_top_v)
+
+            temp_clients.await_event(f'clients={static_clients_num + temp_client}',
+                                     timeout_sec=80,
+                                     from_the_beginning=True,
+                                     backoff_sec=1)
+
+            time.sleep(client_work_time)
+            stop_service_nodes(temp_clients, kill_temp_nodes)
+
+            current_top_v += temp_client
+
+        static_clients.await_event(f'ver={current_top_v}, locNode=', timeout_sec=80,
+                                   from_the_beginning=True, backoff_sec=0.1)
+        static_clients.stop()
+
+        check_topology(control_utility, fin_top_ver)
+
+
+def stop_service_nodes(service: IgniteApplicationService, kill_nodes):
+    """
+    Base service stop command.
+    If kill_nodes=True kill node.
+    If kill_nodes=False then the node is shutting down correctly
+    """
+    if kill_nodes:
+        for node in service.nodes:
+            service.stop_node(node=node, clean_shutdown=False)
+    else:
+        service.stop(clean_shutdown=True)
```

Comment:
Why do not replace this if-else block with `service.stop(clean_shutdown=kill_nodes)`?

Reply:
the base timeout of 10 seconds was not enough to register completion in the base method. However, when they are completed sequentially, I fit into the timeout

Reply:
It's because service.stop waits for completion message "IGNITE_APPLICATION_FINISHED" in any case. So if clean_shutdown is False there is no reason to wait the message. So. It's a bug of `stop` method. Let's fix it instead

Reply:
I guessed, but I was even more afraid to rule. Okay, I'll fix it.